nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1426 @@
+        for i in range(len(self._words)):
+            word = self._words[i]
+            if (
+                word != ""
+            ):  # note that using the current alignments the first word is always empty, so there is no error from indexing the array with i-1
+                align1 = float(
+                    round(self._alignments[i - 1] + start, self._params.data_simulator.outputs.output_precision)
+                )
+                align2 = float(
+                    round(
+                        self._alignments[i] - self._alignments[i - 1],
+                        self._params.data_simulator.outputs.output_precision,
+                    )
+                )
+                text = f"{session_name} {speaker_id} {align1} {align2} {word} 0\n"
+                arr.append((align1, text))
+        return arr
+
+    def create_base_manifest_ds(self) -> str:
+        """
+        Create base diarization manifest file for online data simulation.
+
+        Returns:
+            self.base_manifest_filepath (str): Path to manifest file
+        """
+        basepath = self._params.data_simulator.outputs.output_dir
+        wav_path = os.path.join(basepath, 'synthetic_wav.list')
+        text_path = os.path.join(basepath, 'synthetic_txt.list')
+        rttm_path = os.path.join(basepath, 'synthetic_rttm.list')
+        ctm_path = os.path.join(basepath, 'synthetic_ctm.list')
+        manifest_filepath = os.path.join(basepath, 'base_manifest.json')
+
+        create_manifest(
+            wav_path,
+            manifest_filepath,
+            text_path=text_path,
+            rttm_path=rttm_path,
+            ctm_path=ctm_path,
+            add_duration=False,
+        )
+
+        self.base_manifest_filepath = manifest_filepath
+        return self.base_manifest_filepath
+
+    def create_segment_manifest_ds(self) -> str:
+        """
+        Create segmented diarization manifest file for online data simulation.
+
+        Returns:
+            self.segment_manifest_filepath (str): Path to manifest file
+        """
+        basepath = self._params.data_simulator.outputs.output_dir
+        output_manifest_filepath = os.path.join(basepath, 'segment_manifest.json')
+        input_manifest_filepath = self.base_manifest_filepath
+        window = self._params.data_simulator.segment_manifest.window
+        shift = self._params.data_simulator.segment_manifest.shift
+        step_count = self._params.data_simulator.segment_manifest.step_count
+        deci = self._params.data_simulator.segment_manifest.deci
+
+        create_segment_manifest(input_manifest_filepath, output_manifest_filepath, window, shift, step_count, deci)
+
+        self.segment_manifest_filepath = output_manifest_filepath
+        return self.segment_manifest_filepath
+
+    def _generate_session(self, idx: int, basepath: str, filename: str, enforce_counter: int = 2):
+        """
+        Generate multispeaker audio session and corresponding label files.
```

Comment:
Generate multispeaker audio session -> Generate a multispeaker audio session (This better to be specifically singular since this function is generating a single session.